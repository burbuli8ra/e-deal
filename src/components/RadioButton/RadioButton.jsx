import React from 'react';
import Styled from './RadioButton.style';

const RadioButton = ({
  children,
  label,
  onClick = () => {},
  value
}) => {
  return (
    <Styled.RadioButton isChecked={label === value} onClick={onClick}>
      <Styled.Bullet />
      {children}
    </Styled.RadioButton>
  );
}

export default RadioButton;