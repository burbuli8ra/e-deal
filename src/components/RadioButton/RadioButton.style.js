import styled from '@emotion/styled/macro';
import { colors } from 'theme';

const Bullet = styled.div`
  background-color: ${colors.lightGrey};
  border-radius: 50%;
  box-shadow: inset 0 0 0 1px ${colors.pastelBlue};
  height: 16px;
  width: 16px;
  position: absolute;
  top: 0;
  left: 0;
`;
Bullet.displayName = 'Bullet';

const RadioButton = styled.div`
  color: ${colors.pastelBlue};
  cursor: pointer;
  font-size: 14px;
  line-height: 16px;
  padding-left: 20px;
  position: relative;
  width: fit-content;
  
  ${({ isChecked }) => isChecked ? `
    ${Bullet} {
      box-shadow: inset 0 0 0 6px ${colors.darkPastelBlue};
    }
  ` : `
    &:hover {
      ${Bullet} {
        box-shadow: inset 0 0 0 2px ${colors.darkPastelBlue};
      }
    }
  `}
`;
RadioButton.displayName = 'RadioButton';

export default {
  Bullet,
  RadioButton
};