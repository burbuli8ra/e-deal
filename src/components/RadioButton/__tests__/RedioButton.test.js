import React from 'react';
import { render } from '@testing-library/react';
import RadioButton from '../RadioButton';

const setup = props => <RadioButton {...props} />;

describe('RadioButton component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      categoryMaxPrice: 10000,
      categoryMinPrice: 1000,
      onMaxPriceUpdate: jest.fn(),
      onMinPriceUpdate: jest.fn()
    };
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});