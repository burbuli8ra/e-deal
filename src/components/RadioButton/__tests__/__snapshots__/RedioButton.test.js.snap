// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ursh8z-RadioButton e1xvq25d1"
  >
    <div
      class="css-10vhthi-Bullet e1xvq25d0"
    />
  </div>
</DocumentFragment>
`;
